dissect/target/plugins/os/windows/dpapi/blob.py

```diff
@@ -1,106 +1,150 @@
 from typing import Optional
+from uuid import UUID
 
-from dissect.target.plugins.os.windows.dpapi import crypto
-from dissect.target.plugins.os.windows.dpapi.types.dpapi import DPAPIBlobStruct
+from dissect.cstruct import cstruct
+
+from dissect.target.plugins.os.windows.dpapi.crypto import (
+    CipherAlgorithm,
+    HashAlgorithm,
+    crypt_session_key_type1,
+    crypt_session_key_type2,
+)
+
+blob_def = """
+struct DPAPIBlob {
+    DWORD   dwVersion;
+    char    provider[16];
+    DWORD   mkVersion;
+    char    guid[16];
+    DWORD   flags;
+    DWORD   descriptionLength;
+    char    description[descriptionLength];
+    DWORD   CipherAlgId;
+    DWORD   keyLen;
+    DWORD   saltLength;
+    char    salt[saltLength];
+    DWORD   strongLength;
+    char    strong[strongLength];
+    DWORD   CryptAlgId;
+    DWORD   hashLen;
+    DWORD   hmacLength;
+    char    hmac[hmacLength];
+    DWORD   cipherTextLength;
+    char    cipherText[cipherTextLength];
+    DWORD   signLength;
+    char    sign[signLength];
+};
+"""
+
+c_blob = cstruct()
+c_blob.load(blob_def)
 
 
 class Blob:
-    """Represents a DPAPI blob"""
+    """Represents a DPAPI blob."""
 
     def __init__(self, data: bytes):
-        self._blob = DPAPIBlobStruct.read(data)
+        self._blob = c_blob.DPAPIBlob(data)
 
         self.version = self._blob.dwVersion
-        self.provider = self._blob.provider
-        self.mkguid = self._blob.guid
+        self.provider = str(UUID(bytes_le=self._blob.provider))
         self.mkversion = self._blob.mkVersion
+        self.guid = str(UUID(bytes_le=self._blob.guid))
         self.flags = self._blob.flags
-        self.description = self._blob.description.decode("utf-16-le").encode("utf-8")
-        self.cipher_algo = crypto.CipherAlgorithm.from_id(self._blob.CipherAlgId)
+        self.description = self._blob.description.decode("utf-16-le")
+        self.cipher_algorithm = CipherAlgorithm.from_id(self._blob.CipherAlgId)
         self.key_len = self._blob.keyLen
-        self.hmac = self._blob.hmac
+        self.salt = self._blob.salt
         self.strong = self._blob.strong
-        self.hash_algo = crypto.HashAlgorithm.from_id(self._blob.CryptAlgId)
+        self.hash_algorithm = HashAlgorithm.from_id(self._blob.CryptAlgId)
         self.hash_len = self._blob.hashLen
-        self.ciphertext = self._blob.cipherText
-        self.salt = self._blob.salt
+        self.hmac = self._blob.hmac
+        self.cipher_text = self._blob.cipherText
 
-        self.blob = self._blob.blob
+        # All the blob data between the version, provider and sign fields
+        self.blob = data[20 : -(self._blob.signLength + 4)]
```

Comment:
I'm slightly worried about this . It's basically hardcoded magic values which (theoretically but super unlikely) could change.  
I don't think that it will ever change, so if you prefer this as opposed to another solution, that's fine

Reply:
Made it more dynamic, does that satisfy? I intend to add a `offsetof` function to cstruct to make this easier/more concise.